app/utils.py

```diff
@@ -1,6 +1,46 @@
 import re
 
 
+def list_pointers(input_data, pointer=None):
+    """
+    Recursive function which lists all available pointers in a json structure
+    :param input_data: the input data to search
+    :param pointer: the current pointer
+    :return: generator of the json pointer paths
+    """
+    if isinstance(input_data, dict) and input_data:
+        for k, v in input_data.items():
+            yield pointer + '/' + k if pointer else '/' + k
+            yield from list_pointers(v, pointer + '/' + k if pointer else '/' + k)
+    elif isinstance(input_data, list) and input_data:
+        for index, item in enumerate(input_data):
+            yield '{}/{}'.format(pointer, index) if pointer else '/{}'.format(index)
+            yield from list_pointers(item, '{}/{}'.format(pointer, index))
+
+
+def compare_schemas(source_schema, target_schema):
+    """
+    Compare the pointers in two json structures and return differences
+    :param source_schema: Structure to identify differences against
+    :param target_schema: Target structure to compare against
+    :return:
+    """
+    source_survey_pointers = [p for p in list_pointers(source_schema)]
```

Comment:
Minor: this could be a bit cleaner if a set was used. i.e. `source_survey_pointers.difference(target_survey_pointers)`

Reply:
Good point, done.